services/src/main/java/io/scalecube/services/ServiceCall.java

```diff
@@ -36,56 +37,69 @@
   private final ServiceRegistry serviceRegistry;
   private final Router router;
 
+  private final Function<ServiceMessage, Object> msgToResp =
+      sm -> sm.hasData() ? sm.data() : EMPTY_RESPONSE_ERROR;
+
+  private final Function<Boolean, Function<Mono<ServiceMessage>, Mono<Object>>> monoRespFn =
```

Comment:
that's not the same and previous ones would have been also uglier if we added `EMPTY_RESPONSE_ERROR` instead of filtering empty ones

Reply:
ok, I got it, why do we need use for it some fields instead of methods?

Reply:
there is no difference

Reply:
Actually, it's better to refactor existing `asMono` and `asFlux`  instead of having exacly same thing but done via some tricky field definition. Why somebody has to go over field  and then dive into some logic. Fields are fields. Agree with @segabriel . What we need INHO is just get rid of that silent damn `.filter(ServiceMessage::hasData)`  and that's it. Possibly with minimal changes.

Reply:
done